tornado/gen.py

```diff
@@ -814,6 +814,9 @@ def multi_future(children, quiet_exceptions=()):
     else:
         keys = None
     children = list(map(convert_yielded, children))
+    # filter out NullFutures, like gen.moment
+    children = [child for child in children
```

Comment:
This is too early to do this filtering. If we remove from `children`, we change the result. The result of `yield [None]` should be `[None]`, not `None` or `[]` (and this becomes more apparent if you have a list containing both None/moment and regular Futures). We need to filter instead in the assignment to `unfinished_children`.

Reply:
oops, right. I updated it to just change the assert to accept NullFutures, since the only function called on them is `future_add_done_callback`, which is safe on NullFutures, which are always done.